src/instructlab/eval/mmlu.py

```diff
@@ -172,19 +191,25 @@ def _run_mmlu(self, server_url: str | None = None) -> dict:
                 raise InvalidTasksDirError(self.tasks_dir)
             tm = TaskManager(verbosity="DEBUG", include_path=self.tasks_dir)
         should_apply_chat_template = self.system_prompt is not None
-        mmlu_output = self._simple_evaluate_with_error_handling(
-            model=model,
-            model_args=model_args,
-            tasks=self.tasks,
-            num_fewshot=self.few_shots,
-            batch_size=self.batch_size,
-            device=self.device,
-            task_manager=tm,
-            system_instruction=self.system_prompt,
-            apply_chat_template=should_apply_chat_template,
+
+        # configure the args here so users can override them as necessary
+        simple_evaluate_kwargs = {
+            "model": model,
+            "model_args": model_args,
+            "tasks": self.tasks,
+            "num_fewshot": self.few_shots,
+            "batch_size": self.batch_size,
+            "device": self.device,
+            "task_manager": tm,
+            "system_instruction": self.system_prompt,
+            "apply_chat_template": should_apply_chat_template,
+        }
+        simple_evaluate_kwargs.update(extra_args)
+
+        self._results = self._simple_evaluate_with_error_handling(
+            **simple_evaluate_kwargs
         )
-        results = mmlu_output["results"]
-        return results
+        return self.results["results"]
```

Comment:
Was this meant to be _results?

Reply:
No, we set the attribute so why shouldn't we read from it? If we face errors in this line then that is indiciative of another problem

Reply:
I ended up changing this line to store and return the reference in a local variable instead. MyPy isn't smart enough to detect that returning the attribute or hidden attribute directly is always going to be non-None